Assignment/Test.c

```diff
@@ -4,11 +4,143 @@
+}
+
+void TestCases () {
+   int* input[] = { (int[]) { 12, 2, 34, 22, 13 }, (int[]) { 6, 23, 20, 12 },
+                    (int[]) { 6, 5, 4, 3, 2, 1 }, (int[]) { 3, 5, 65 },
+                    (int[]) { 3, 3, 3, 3, 3, 3 }, (int[]) { -2, -5, -10, 10, -3, -5 } },
+     * expOutput[] = { (int[]) { 2, 12, 13, 22, 34 }, (int[]) { 6, 12, 20, 23 },
+                       (int[]) { 1, 2, 3, 4, 5, 6 }, (int[]) { 3, 5, 65 },
+                       (int[]) { 3, 3, 3, 3, 3, 3 }, (int[]) { -10, -5, -5, -3, -2, 10 } };
+   int numOfTestCases = (int)sizeof (input) / sizeof (input[0]), size[] = { 5, 4, 6, 3, 6, 6 };
+   printf ("\n\t\t\t\tTestcases for Insertion Sort\n\n"
+           "\tInput\t\t\t\tOutput\t\t\tTest Case Result\n |--------------------------|"
+           "--------------------------|--------------------------|\n");
+   for (int i = 0; i < numOfTestCases; i++) {
+      int* actualOutput = malloc (size[i] * sizeof (int));
+      if (actualOutput == NULL) {
+         fprintf (stderr, "memory allcoation failed");
+         return;
+      }
+      for (int j = 0; j < size[i]; j++) actualOutput[j] = input[i][j];
+      InsertionSort (actualOutput, size[i]);
+      PrintTable (input[i], size[i], 6);
+      PrintTable (expOutput[i], size[i], 6);
+      const char* result = CompareArr (actualOutput, expOutput[i], size[i]) ?
+         GREEN_TEXT "Pass" RESET_TEXT : RED_TEXT "fail" RESET_TEXT;
+      int padding = (30 - strlen (result)) / 2; // Calculate left padding
+      printf (" |%*s%20s%*s   |\n", padding, "", result, padding, "");
+      free (actualOutput);
+   }
+   printf ("\n\t\t\t\tTest Cases for Binary Search\n\n"
+           "\tInput\t\t\t\tOutput\t\t  Key\t\tResult\n |--------------------------|"
+           "--------------------------|------|------------------------|\n");
+   for (int i = 0; i < numOfTestCases; i++) {
+      int* actualOutput = malloc (size[i] * sizeof (int));
+      PrintTable (input[i], size[i], 6);
+      PrintTable (expOutput[i], size[i], 6);
+      int Keys[] = { 3, 25, -1, 65, 3, -10 },
+         foundIndex = BinarySearch (expOutput[i], size[i], Keys[i]);
+      printf (" | %-3d  | ", Keys[i]);
+      printf (foundIndex != -1 ? GREEN_TEXT "Element %3d found at %d" RESET_TEXT
+              : RED_TEXT "Element %3d Not Found " RESET_TEXT, Keys[i], foundIndex);
+      printf (" |\n");
+      free (actualOutput);
+   }
+}
+
+void UserInput () {
+   int arr[100] = { 0 }, n;
+   char buffer[256], * endptr;
+   while (1) {
+      printf ("Enter number of elements: ");
+      fgets (buffer, sizeof (buffer), stdin);
+      n = strtol (buffer, &endptr, 10);
+      // Check for valid input
+      if (endptr != buffer && *endptr == '\n' && n > 0 && n <= 100) break; // Valid input,exit loop
+   }
+   for (int i = 0; i < n; i++) {
+      printf ("Enter element %d: ", i + 1);
+      fgets (buffer, sizeof (buffer), stdin);
+      arr[i] = strtol (buffer, &endptr, 10);
+      if (endptr == buffer || *endptr != '\n') i--;

```

Comment:
Since you are repeating this bit of code every time you need user input, you can declare this as a separate function.